test/curl/httppost.js

```diff
@@ -121,3 +122,42 @@ it( 'should upload file correctly', function ( done ) {
     curl.perform();
 
 });
+
+it( 'should upload file correctly with filename', function ( done ) {
+
+    var _name = Math.random().toString(36).substring(7);
+
+    var postData = [{
+        name     : 'file',
+        file     : imageFilename,
+        type     : 'image/png',
+        filename : _name
+    }];
+    
+    curl.setOpt( 'HTTPPOST', postData );
+
+    curl.on( 'end', function( status, data ) {
+
+        if ( status !== 200 ) {
+            throw Error( 'Invalid status code: ' + status );
+        }
+
+        data = JSON.parse( data );
+
+        data.size.should.be.equal( size );
+        //data.name.should.be.equal( imageFilename.split( path.sep ).pop() );
+        data.type.should.be.equal( postData[0].type );
+        data.filename.should.be.equal( _name );
```

Comment:
Remove this line.

``data`` here is the one returned on line [:71](https://github.com/SamuelePilleri/node-libcurl/blob/develop/test/curl/httppost.js#L71). So you just need to check for the name field to be equals the _name variable (line 148 above that you commented).